augur/mask.py

```diff
@@ -105,13 +107,24 @@ def mask_sequence(sequence, mask_sites, mask_from_beginning, mask_from_end, mask
     sequence_length = len(sequence.seq)
     beginning, end = mask_from_beginning, mask_from_end
 
+    if mask_gaps == "terminals":
+        # Count leading and trailing gaps
+        leading_gaps = len(sequence.seq) - len(sequence.seq.lstrip('-'))
+        trailing_gaps = len(sequence.seq) - len(sequence.seq.rstrip('-'))
+        # Update beginning and end to account for leading and trailing gaps
+        beginning = max(beginning, leading_gaps)
+        end = max(end, trailing_gaps)
+
     if beginning + end > sequence_length:
         beginning, end = sequence_length, 0
 
     seq = str(sequence.seq)[beginning:-end or None]
 
     if mask_invalid:
         seq = "".join(nuc if nuc in VALID_NUCLEOTIDES else "N" for nuc in seq)
+    
+    if mask_gaps == "all":
+        seq = "".join(nuc if nuc != "-" else "N" for nuc in seq)

```

Comment:
_(non-blocking)_

```suggestion
        seq = "".join("N" if nuc == "-" else nuc for nuc in seq)
```

This reads slightly better to me - it pairs the mask character with the gap character.